gatsby-node.js

```diff
@@ -59,51 +59,73 @@ exports.createPages = ({ actions }) => {
-//   plugins: [
-//     {
-//       name: "esdoc-ecmascript-proposal-plugin",
-//       option: {
-//         classProperties: true,
-//         objectRestSpread: true,
-//         doExpressions: true,
-//         functionBind: true,
-//         functionSent: true,
-//         asyncGenerators: true,
-//         decorators: true,
-//         exportExtensions: true,
-//         dynamicImport: true,
-//       },
-//     },
-//     { name: "esdoc-accessor-plugin" },
-//   ],
-// }
-// exports.createPages = async ({ actions: { createPage } }) => {
-//   esdoc.generate(ESDOC_CONFIG)
-//   console.log(fs)
-//   console.log(esdoc)
-//   // debugger
-//
-//   // Example from Gatsby docs
-//   // `getPokemonData` is a function that fetches our data
-//   // const allPokemon = await getPokemonData(["pikachu", "charizard", "squirtle"])
-//   // Create a page that lists all Pokémon.
-//   // createPage({
-//   //   path: `/`,
-//   //   component: require.resolve("./src/templates/all-pokemon.js"),
-//   //   context: { allPokemon },
-//   // })
-//   //
-//   // // Create a page for each Pokémon.
-//   // allPokemon.forEach(pokemon => {
-//   //   createPage({
-//   //     path: `/pokemon/${pokemon.name}/`,
-//   //     component: require.resolve("./src/templates/pokemon.js"),
-//   //     context: { pokemon },
-//   //   })
-//   // })
-// }
+// DOC STUFF TODO Extract
+
+let esdoc = require("esdoc").default
+let tmp = require("tmp")
+let slugify = require("@sindresorhus/slugify")
+
+let generateESDoc = function(config) {
+  var tmpdir = tmp.dirSync()
+  esdoc.generate({ ...config, ...{ destination: tmpdir.name } })
+  let index = fs.readFileSync(`${tmpdir.name}/index.json`)
+  let result = JSON.parse(index)
+  tmpdir.removeCallback()
+
+  return result
+}
+
+exports.sourceNodes = async ({
+  actions: { createNode },
+  createNodeId,
+  createContentDigest,
+}) => {
+  let docNodes = generateESDoc({
+    source: "./node_modules/@miragejs/server/lib",
+    excludes: ["(node_modules|tests|tmp)"],
+    plugins: [
+      {
+        name: "esdoc-ecmascript-proposal-plugin",
+        option: {
+          classProperties: true,
+          objectRestSpread: true,
+          doExpressions: true,
+          functionBind: true,
+          functionSent: true,
+          asyncGenerators: true,
+          decorators: true,
+          exportExtensions: true,
+          dynamicImport: true,
+        },
+      },
+      { name: "esdoc-accessor-plugin" },
+    ],
+  })
+
+  docNodes.forEach(docNode => {
+    let node = {
+      ...{ slug: slugify(docNode.name) },
+      ...docNode,
+    }
+
+    let data = {
+      ...node,
+      ...{
+        id: createNodeId(`esdoc-${node.__docId__}`),
+        internal: {
+          type: "ESDoc",
+          contentDigest: createContentDigest(JSON.stringify(node)),
+        },
+      },
+    }
+
+    createNode(data)
+
+    if (node.kind === "class" && node.access === "public") {
+      console.log("TODO create ssr page for", node.name)
```

Comment:
Thought: could we put all routes into Content Mesh, potentially as a new type that we can then query in both node and browser? Then instead of creating pages here we can just query all routes, iterate through them and create pages elsewhere?

Reply:
Yes I think this is where we'll end up.